example_test.go

```diff
@@ -358,3 +361,46 @@ func ExampleWrapCore_wrap() {
 	// {"level":"info","msg":"doubled"}
 	// {"level":"info","msg":"doubled"}
 }
+
+type timeouts struct {
+	readTimeout  time.Duration
+	writeTimeout time.Duration
+}
+
+func parseTimeouts(line string) (timeouts, error) {
+	parts := strings.Split(line, " ")
+	if len(parts) != 2 {
+		return timeouts{}, errors.New("line not in expected format")
+	}
+
+	read, err := time.ParseDuration(parts[0])
+	if err != nil {
+		return timeouts{}, zap.WrapError(fmt.Errorf("parse readTimeout: %w", err),
+			zap.String("readTimeout", parts[0]))
+	}
+
+	write, err := time.ParseDuration(parts[1])
+	if err != nil {
+		return timeouts{}, zap.WrapError(fmt.Errorf("parse writeTimeout: %w", err),
```

Comment:
Because the example will be used as reference, maybe we should reduce redundancy in the message here. The log message already says we're parsing a timeout, so maybe just the fields will be enough here.

```suggestion
		return timeouts{}, zap.WrapError(err, zap.String("readTimeout", parts[0]))
	}

	write, err := time.ParseDuration(parts[1])
	if err != nil {
		return timeouts{}, zap.WrapError(err,
```

Theoretically, if there was a caller to this function before the log, it would add the `parseTimeouts: ` prefix, and that would also make it to the log.